File: demo/components/dropdown/dropdown-demo.ts
import {Component} from 'angular2/core';
import {CORE_DIRECTIVES} from 'angular2/common';
import {DROPDOWN_DIRECTIVES} from '../../../ng2-bootstrap';

// webpack html imports
let template = require('./dropdown-demo.html');

@Component({
  selector: 'dropdown-demo',
  directives: [DROPDOWN_DIRECTIVES, CORE_DIRECTIVES],
  template: template
})
export class DropdownDemoComponent {
  public disabled:boolean = false;
  public status:{isopen:boolean} = {isopen: false};
  public items:Array<string> = ['The first choice!',
    'And another choice for you.', 'but wait! A third!'];

  public toggled(open:boolean):void {
    console.log('Dropdown is now: ', open);
  }

  public toggleDropdown($event:MouseEvent):void {
    $event.preventDefault();
    $event.stopPropagation();
    this.status.isopen = !this.status.isopen;
  }
}

File: web/components/dropdown/dropdown-demo.dart
import "package:angular2/angular2.dart";
import 'package:ng2-strap/dropdown/index.dart';
import 'dart:html';

@Component(selector: "dropdown-demo",
    templateUrl: 'dropdown-demo.html',
    directives: const [DROPDOWN_DIRECTIVES, CORE_DIRECTIVES])
class DropdownDemo {
  bool disabled = false;
  Map status = {"isopen": false};
  List<String> items = [
    "The first choice!",
    "And another choice for you.",
    "but wait! A third!"
  ];
  void toggled(bool open) {
    print("Dropdown is now: $open");
  }

  void toggleDropdown(MouseEvent $event) {
    $event.preventDefault();
    $event.stopPropagation();
    this.status['isopen'] = !this.status['isopen'];
  }
}
